k8s-operator/apis/v1alpha1/types_tsdnsconfig.go

```diff
@@ -85,6 +85,11 @@ type Nameserver struct {
 	// Service configuration.
 	// +optional
 	Service *NameserverService `json:"service,omitempty"`
+	// Replicas specifies how many devices to create. Set this to enable
+	// high availability for nameservers. Defaults to 1.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
Just comparing this with ProxyClass and the other PR you have open; I think maybe we should put this under a `deployment` field for consistency and future additions. Similar to how we have a `service` field at this level, and then service-specific config under it.

Reply:
I think after our chat we came to the conclusion that it is fine here for now and when we move to beta types we can move things around. Just so we have image/tag also at a more sensible location within the crd.